pkg/reconciler/certificate/certificate.go

```diff
@@ -0,0 +1,229 @@
+	err = c.reconcile(ctx, knCert)
+	if equality.Semantic.DeepEqual(original.Status, knCert.Status) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the informer's
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+	} else if _, err := c.updateStatus(knCert); err != nil {
+		logger.Warnw("Failed to update certificate status", zap.Error(err))
+		c.Recorder.Eventf(knCert, corev1.EventTypeWarning, "UpdateFailed",
+			"Failed to update status for Certificate %s: %v", key, err)
+		return err
+	}
+	if err != nil {
+		c.Recorder.Event(knCert, corev1.EventTypeWarning, "InternalError", err.Error())
+	}
+	return err
+}
+
+func (c *Reconciler) reconcile(ctx context.Context, knCert *v1alpha1.Certificate) error {
+	logger := logging.FromContext(ctx)
+
+	knCert.SetDefaults(ctx)
+	knCert.Status.InitializeConditions()
+
+	logger.Info("Reconciling Cert-Manager certificate for Knative cert %s/%s.", knCert.Namespace, knCert.Name)
+	cmConfig := config.FromContext(ctx).CertManager
+	cmCert := resources.MakeCertManagerCertificate(cmConfig, knCert)
+	cmCert, err := c.reconcileCMCertificate(ctx, knCert, cmCert)
+	if err != nil {
+		return err
+	}
+
+	knCert.Status.NotAfter = cmCert.Status.NotAfter
+	knCert.Status.ObservedGeneration = knCert.Generation
+	// Propagate cert-manager Certificate status to Knative Certificate.
+	cmCertReadyCondition := resources.GetReadyCondition(cmCert)
+	if cmCertReadyCondition == nil {
+		return nil
+	}
+	switch {
+	case cmCertReadyCondition.Status == cmv1alpha1.ConditionUnknown:
+		knCert.Status.MarkUnknown(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
+	case cmCertReadyCondition.Status == cmv1alpha1.ConditionTrue:
+		knCert.Status.MarkReady()
+	case cmCertReadyCondition.Status == cmv1alpha1.ConditionFalse:
+		knCert.Status.MarkNotReady(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
+	}
+	return nil
+}
+
+func (c *Reconciler) reconcileCMCertificate(ctx context.Context, knCert *v1alpha1.Certificate, desired *cmv1alpha1.Certificate) (*cmv1alpha1.Certificate, error) {
+	logger := logging.FromContext(ctx)
+	cmCert, err := c.cmCertificateLister.Certificates(desired.Namespace).Get(desired.Name)
+	if apierrs.IsNotFound(err) {
+		cmCert, err = c.certManagerClient.CertmanagerV1alpha1().Certificates(desired.Namespace).Create(desired)
+		if err != nil {
+			logger.Errorw("Failed to create Cert-Manager certificate", zap.Error(err))
+			c.Recorder.Eventf(knCert, corev1.EventTypeWarning, "CreationFailed",
+				"Failed to create Cert-Manager Certificate %s/%s: %v", desired.Name, desired.Namespace, err)
+			return nil, err
+		}
+		c.Recorder.Eventf(knCert, corev1.EventTypeNormal, "Created",
+			"Created Cert-Manager Certificate %s/%s", desired.Namespace, desired.Name)
+	} else if err != nil {
+		return nil, err
+	} else if !equality.Semantic.DeepEqual(cmCert.Spec, desired.Spec) {
```

Comment:
I think this is missing the `NotOwned` checks that we added to our reconcile pattern.  Look for that string or `IsControlledBy` in our other controllers.